runtime/runtime/src/tests/apply.rs

```diff
@@ -2793,3 +2837,169 @@ fn test_deploy_and_call_local_receipts() {
+        8,
+        "should have processed 5 transactions and 3 local receipts"
+    );
+    let tx_outcomes = apply_result.outcomes.iter().take(5).map(|o| o.id).collect::<Vec<_>>();
+    assert_eq!(tx_outcomes, expected_order, "outcomes are not in expected sorted order");
+}
+
+#[test]
+fn test_transaction_multiple_access_keys_with_apply() {
+    let alice_key1 = InMemorySigner::from_seed(alice_account(), KeyType::ED25519, "seed1");
+    let alice_key2 = InMemorySigner::from_seed(alice_account(), KeyType::ED25519, "seed2");
+    let alice_key3 = InMemorySigner::from_seed(alice_account(), KeyType::ED25519, "seed3");
+    let bob_key = InMemorySigner::from_seed(bob_account(), KeyType::ED25519, "seed4");
+
+    let send_money_tx = |nonce, key| {
+        SignedTransaction::send_money(
+            nonce,
+            alice_account(),
+            bob_account(),
+            key,
+            to_yocto(1000),
+            CryptoHash::default(),
+        )
+    };
+
+    let txs = vec![
+        send_money_tx(1, &alice_key1),
+        send_money_tx(1, &alice_key2),
+        send_money_tx(1, &alice_key3),
+        send_money_tx(2, &alice_key3),
+        send_money_tx(2, &alice_key1),
+        send_money_tx(3, &alice_key1),
+    ];
+
+    let accounts_with_keys = vec![
+        (
+            alice_account(),
+            vec![
+                Arc::new(alice_key1.clone()),
+                Arc::new(alice_key2.clone()),
+                Arc::new(alice_key3.clone()),
+            ],
+        ),
+        (bob_account(), vec![Arc::new(bob_key.clone())]),
+    ];
+
+    let (runtime, tries, root, mut apply_state, _signers, epoch_info_provider) =
+        setup_runtime_with_keys(
+            accounts_with_keys,
+            to_yocto(1_000_000),
+            to_yocto(500_000),
+            10u64.pow(15),
+        );
+
+    let flags = vec![true; txs.len()];
+    let s_valid_txs = SignedValidPeriodTransactions::new(&txs, flags.as_slice());
+    let apply_result = runtime
+        .apply(
+            tries.get_trie_for_shard(ShardUId::single_shard(), root),
+            &None,
+            &apply_state,
+            &[],
+            s_valid_txs,
+            &epoch_info_provider,
+            Default::default(),
+        )
+        .expect("apply should succeed");
+
+    let expected_order = txs.iter().map(|tx| tx.get_hash()).collect::<Vec<_>>();
+
+    assert_eq!(apply_result.outcomes.len(), 6, "should have processed 6 transactions");
+    let tx_outcomes = apply_result.outcomes.iter().map(|o| o.id).collect::<Vec<_>>();
+    assert_eq!(tx_outcomes, expected_order, "outcomes are not in expected sorted order");
+
+    let shard_uid = ShardUId::single_shard();
+    let root = commit_apply_result(&apply_result, &mut apply_state, &tries, shard_uid);
+    let state = tries.new_trie_update(shard_uid, root);
+    let account = get_account(&state, &alice_account()).unwrap().unwrap();
+
+    assert!(account.amount() < to_yocto(994_000));
```

Comment:
Perhaps this could be improved to `assert_eq!(account.amount(), exact_expected_amount_after_fees)` but I don't know how to calculate them to exactly match. And maybe it's not worth it for the purposes of this test